packages/hub/src/lib/LazyBlob.ts

```diff
@@ -0,0 +1,92 @@
+import { createReadStream } from "node:fs";
+import { open } from "node:fs/promises";
+import type { FileHandle } from "node:fs/promises";
+import { Readable } from "node:stream";
+
+export class LazyBlob extends Blob {
+	static async create(path: string, start?: number, end?: number): Promise<LazyBlob> {
+		const lazyBlob = new LazyBlob(path, start, end);
+
+		await lazyBlob.init();
+
+		return lazyBlob;
+	}
```

Comment:
```suggestion
	static async create(path: string): Promise<LazyBlob> {
		const { size } = await stat(path);
		
		const lazyBlob = new LazyBlob(path, 0, size);

		return lazyBlob;
	}
```

That way `LazyBlob.start` and `LazyBlob.end` will always be defined. It'll remove the need for `LazyBlob.totalSize` too.